framework/mutex/OracleMutex.php

```diff
@@ -0,0 +1,131 @@
+<?php
+/**
+ * @link http://www.yiiframework.com/
+ * @copyright Copyright (c) 2008 Yii Software LLC
+ * @license http://www.yiiframework.com/license/
+ */
+
+namespace yii\mutex;
+
+use PDO;
+use Yii;
+use yii\base\InvalidConfigException;
+
+/**
+ * OracleMutex implements mutex "lock" mechanism via Oracle locks.
+ *
+ * Application configuration example:
+ *
+ * ```
+ * [
+ *     'components' => [
+ *         'db' => [
+ *             'class' => 'yii\db\Connection',
+ *             'dsn' => 'oci:dbname=LOCAL_XE',
+ *              ...
+ *         ]
+ *         'mutex' => [
+ *             'class' => 'yii\mutex\OracleMutex',
+ *              ...
+ *         ],
+ *     ],
+ * ]
+ * ```
+ *
+ * @see Mutex
+ *
+ * @author Alexander Zlakomanov <zlakomanoff@gmail.com>
+ * @since 1.0
+ */
+class OracleMutex extends DbMutex
+{
+    /**
+     * @var string driver name.
+     */
+    public $driverName = 'oci';
+
+    /**
+     * @var array available lock modes
+     */
+    private $availableLockModes = [
+        'X_MODE',
+        'NL_MODE',
+        'S_MODE',
+        'SX_MODE',
+        'SS_MODE',
+        'SSX_MODE'
+    ];
```

Comment:
In this case restriction should be removed completely: there is no sense to keep this array if its value can not be used anyhow.
It is better to list all available options at PhpDoc for `lockMode` instead.